ApplicationLibCode/ProjectDataModel/RimFileWellPath.cpp

```diff
@@ -275,3 +294,58 @@ void RimFileWellPath::updateFilePathsFromProjectPath( const QString& newProjectP
         m_filePathInCache = newCacheFileName;
     }
 }
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimFileWellPath::ensureWellPathStartAtSeaLevel( RigWellPath* wellPath )
+{
+    std::vector<cvf::Vec3d> wellPathPoints = wellPath->wellPathPoints();
+    std::vector<double>     measuredDepths = wellPath->measuredDepths();
+    if ( wellPathPoints.empty() || measuredDepths.empty() || wellPathPoints.size() != measuredDepths.size() ) return;
+
+    cvf::Vec3d   firstPoint           = wellPathPoints[0];
+    const double epsilon              = 1e-3;
+    bool         firstPointAtSeaLevel = std::abs( firstPoint.z() ) < epsilon;
+    if ( !firstPointAtSeaLevel )
+    {
+        // Insert a new point on sea level straight above the
+        // first point in the read well path.
+        std::vector<cvf::Vec3d> newPoints = { cvf::Vec3d( firstPoint.x(), firstPoint.y(), 0.0 ) };
+        newPoints.insert( newPoints.end(), wellPathPoints.begin(), wellPathPoints.end() );
+        wellPath->setWellPathPoints( newPoints );
+
+        // Give the new point zero measured depth.
+        std::vector<double> newMeasuredDepths = { 0.0 };
+
+        // And adjust measured depths of the read data
+        // with the diff between the old and new first point.
+        double diffMd = std::abs( newPoints[1].z() - newPoints[0].z() );
+        for ( double originalMd : measuredDepths )
+        {
+            newMeasuredDepths.push_back( originalMd + diffMd );
+        }
+
+        wellPath->setMeasuredDepths( newMeasuredDepths );
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimFileWellPath::fieldChangedByUi( const caf::PdmFieldHandle* changedField,
+                                        const QVariant&            oldValue,
+                                        const QVariant&            newValue )
+{
+    RimWellPath::fieldChangedByUi( changedField, oldValue, newValue );
+
+    if ( changedField == &m_useAutoGeneratedPointAtSeaLevel )
+    {
+        std::unique_ptr<RifWellPathImporter> wellPathImporter = std::make_unique<RifWellPathImporter>();
```

Comment:
I this can be simplified to

    RifWellPathImporter wellPathImporter;
    QString                              errorMessage;
    if ( readWellPathFile( &errorMessage, &wellPathImporter, false ) )

Reply:
Fixed!